src/main/java/edu/wpi/grip/core/operations/composite/FilterContoursOperation.java

```diff
@@ -0,0 +1,110 @@
+package edu.wpi.grip.core.operations.composite;
+
+import com.google.common.eventbus.EventBus;
+import edu.wpi.grip.core.InputSocket;
+import edu.wpi.grip.core.Operation;
+import edu.wpi.grip.core.OutputSocket;
+import edu.wpi.grip.core.SocketHint;
+
+import java.io.InputStream;
+import java.util.Optional;
+
+import static edu.wpi.grip.core.operations.composite.FindContourOperation.Contours;
+import static org.bytedeco.javacpp.opencv_core.*;
+import static org.bytedeco.javacpp.opencv_imgproc.*;
+
+/**
+ * An {@link Operation} that takes in a list of contours and outputs a list of any contours in the input that match
+ * all of several criteria.  Right now, the user can specify a minimum area, minimum perimeter, and ranges for width
+ * and height.
+ * <p>
+ * This is useful because running a Find Contours on a real-life image typically leads to many small undesirable
+ * contours from noise and small objects, as well as contours that do not meet the expected characteristics of the
+ * feature we're actually looking for.  So, this operation can help narrow them down.
+ */
+public class FilterContoursOperation implements Operation {
+
+    private final SocketHint<Contours> contoursHint = new SocketHint<Contours>("Contours", Contours.class, Contours::new);
+    private final SocketHint<Number> minAreaHint = new SocketHint<>("Min Area", Number.class, 0, SocketHint.View.SPINNER);
+    private final SocketHint<Number> minPerimeterHint = new SocketHint<>("Min Perimeter", Number.class, 0,
+            SocketHint.View.SPINNER);
+    private final SocketHint<Number> minWidthHint = new SocketHint<>("Min Width", Number.class, 0,
+            SocketHint.View.SPINNER);
+    private final SocketHint<Number> maxWidthHint = new SocketHint<>("Max Width", Number.class, Integer.MAX_VALUE,
```

Comment:
:-1: You realize `Integer.MAX_VALUE` is like 2 billion right?  It would be impossible to select a reasonable range (which I think would probably be around 1,000-100,000)